server/server.go

```diff
@@ -54,32 +103,22 @@ type RPCServer struct {
 	// bus.
 	consumeSettings connection.ConsumeSettings
 
-	// publishSettings is the configuration used when publishing a message with
-	// the client
-	publishSettings connection.PublishSettings
-}
-
-// processedRequest is used to add the response from a handler func combined
-// with a amqp.Delivery. The reasone we need to combine those is that we reply
-// to each request in a separate go routine and the delivery is required to
-// determine on which queue to reply.
-type processedRequest struct {
-	delivery amqp.Delivery
-	response []byte
+	// ctx is the basic context for the server. This is used to gracefully stop and handle timeouts.
+	ctx context.Context
```

Comment:
I know this is some kind of best practice but is this the best way for us? Right now we will only hold one value in this context, right? I don't really have a strong feeling about this but I've been f*cked by `context.Context` before.

Reply:
Yep, I understand and I agree with you. From the documentation:

> Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions.

But to your point the documentation also says:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

That means that we should create the first `context.Background()` inside the consumer. I'll have to think about how to signal the running handlers that they should stop when we want to stop the server.